OptimizelySDK/ProjectConfig.cs

```diff
@@ -353,5 +359,90 @@ public Variation GetVariationFromId(string experimentKey, string variationId)
+
+            Dictionary<string, string> experimentToVariationMap = _ForcedVariationMap[userId];
+
+            string experimentId = GetExperimentFromKey(experimentKey).Id;
+
+            // this case is logged in getExperimentFromKey
+            if (string.IsNullOrEmpty(experimentId))
+                return null;
+            
+            if (experimentToVariationMap.ContainsKey(experimentId) == false)
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format(@"No experiment ""{0}"" mapped to user ""{1}"" in the forced variation map.", experimentKey, userId));
+                return null;
+            }
+            
+            string variationId = experimentToVariationMap[experimentId];
+
+            if (string.IsNullOrEmpty(variationId))
+            {
+                Logger.Log(LogLevel.DEBUG, string.Format(@"No variation mapped to experiment ""{0}"" in the forced variation map.", experimentKey));
+                return null;
+            }
+            
+            string variationKey = GetVariationFromId(experimentKey, variationId).Key;
+
+            // this case is logged in getVariationFromKey
+            if (string.IsNullOrEmpty(variationKey))
+                return null;
+
+            Logger.Log(LogLevel.DEBUG, string.Format(@"Variation ""{0}"" is mapped to experiment ""{1}"" and user ""{2}"" in the forced variation map", variationKey, experimentKey, userId));
+            
+            Variation variation = GetVariationFromKey(experimentKey, variationKey);
+
+            return variation;
+        }
+        
+        /// <summary>
+        /// Sets an associative array of user IDs to an associative array of experiments to forced variations.
+        /// </summary>
+        /// <param name="experimentKey">key for Experiment</param>
+        /// <param name="userId">The User ID</param>
+        /// <param name="variationKey">key for Variation</param>
+        /// <returns>A boolean value that indicates if the set completed successfully.</returns>
+        public bool SetForcedVariation(string experimentKey, string userId, string variationKey)
+        {
+            var experimentId = GetExperimentFromKey(experimentKey).Id;
+
+            // this case is logged in getExperimentFromKey
+            if (string.IsNullOrEmpty(experimentId))
+                return false;
+
+            if (string.IsNullOrEmpty(variationKey))
+            {
+                _ForcedVariationMap.Remove(userId);
+                Logger.Log(LogLevel.DEBUG, string.Format(@"Variation mapped to experiment ""{0}"" has been removed for user ""{1}"".", experimentKey, userId));
+                return true;
+            }
+            
+            string variationId = GetVariationFromKey(experimentKey, variationKey).Id;
+
+            // this case is logged in getVariationFromKey
+            if (string.IsNullOrEmpty(variationId))
+                return false;
+
+            _ForcedVariationMap[userId] = new Dictionary<string, string> {
```

Comment:
This won't work if the user ID has multiple experiment-to-variation mapping since you're not appending to that user ID mapping; it's getting overwritten each time. You have to support this data structure:

`
{
"userId1" : { 	"experimentId1" : "variationId1a",
	       	"experimentId2" : "variationId2",
	        "experimentId3" : "variationId3" },
"userId2" : { 	"experimentId1" : "variationId1b" }
}
`